[hipsparse] Improve testing coverage for bsrxmv routine in hipSPARSE (#3576)
ROCm/rocm-libraries: projects/hipsparse/CHANGELOG.md

* In `hipsparseSpSM_solve()`, the external buffer is passed as a parameter, which does not match the NVIDIA CUDA cuSPARSE API. The `hipsparseSpSM_solve_ex()` routine has been added to properly match the
NVIDIA CUDA cuSPARSE `cusparseSpSM_solve()` API. The original `hipsparseSpSM_solve()` routine has been 
deprecated and will be removed in a future release.

> * Fix bug where we sometimes read memory out of bounds in the single precision bsrxmv kernels when `block_dim` equals `5` or `8`.

Reviewer: ```suggestion
* Fixed an issue where out-of-bounds memory reads can occur in the single precision bsrxmv kernels when `block_dim` equals `5` or `8`.
```
